dev/com.ibm.ws.microprofile.openapi/src/com/ibm/ws/microprofile/openapi/ApplicationProcessor.java

```diff
@@ -99,6 +104,15 @@ public void activate(ComponentContext cc) {
         } else {
             this.document = createBaseOpenAPIDocument();
         }
+
+        if(openAPIEndpointTracker == null){
+            BundleContext bundleContext = cc.getBundleContext();
+            // make sure we have a context before trying to use it to create the tracker for the OpenAPI doc endpoint
+            if(bundleContext != null) {
+                openAPIEndpointTracker = new ServiceTracker<>(bundleContext, OpenAPIEndpointProvider.class, null);
+                openAPIEndpointTracker.open();
+            }
+        }
```

Comment:
Because this is a component, we should use an `@Reference` annotated field to get the `OpenAPIEndpointProvider`, rather than a tracker.